### DISCUSSION TITLE
How to efficiently batch multiple instances of the same graph (only difference being node embeddings)

### DISCUSSION THREAD

**Discussion by jasperhyp** [pyg-team/pytorch_geometric#6247] [Q&A] [answered]
Hi!  I have one graph and a number of different sets of node embeddings for this graph.  Using a larger adjacency matrix does not sound efficient in this case.  I think the simplest way is to feed in 3D node embedding input (`[batch_size, node_num, feat_dim]`), so I experimented and it looks like GATv2Conv does not allow for 3D node embeddings.  How should I deal with this?

**Comment by rusty1s** [comment, marked as answer]
Yes, not all GNN operators can operate on 3D tensors. It is currently recommended to follow the standard PyG batching in other cases. This will solely replicate the `edge_index` for each example in the mini-batch once, which is negligible in terms of memory consumption.

**Reply by jasperhyp** [reply]
Ok cool, thank you!
